### PR TITLE
Add Mujoco v5 environments

### PR DESCRIPTION
**Pull request by LucasAlegre**
This PR should be merged when Gymnasium v1.0 is out. It should add:

- mo-ant-v5
- mo-halfcheetah-v5
- mo-hopper-v5
- mo-swimmer-v5
- mo-walker2d-v5
- mo-humanoid-v5

### CODE REVIEW

**Comment by ffelten** [comment]
No swimmer as in #39  ?

**Comment by LucasAlegre** [comment]
> No swimmer as in #39 ?

True, I forgot about this one.

**Review by Kallinteris-Andreas — COMMENTED** [review]
HalfCheetah LGTM

Hopper needs docstring update

File: mo_gymnasium/envs/mujoco/ant_v5.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    ## Version History
+    - v5: Now includes contact forces in the reward and observation.
+    See https://gymnasium.farama.org/environments/mujoco/ant/#version-history
+    """
+
+    def __init__(self, cost_objective=True, **kwargs):
+        super().__init__(**kwargs)
+        EzPickle.__init__(self, cost_objective, **kwargs)
+        self.cost_objetive = cost_objective
```
Comment (Kallinteris-Andreas) [inline mo_gymnasium/envs/mujoco/ant_v5.py:35]: `self._cost_objective =...` As it is not intended to be user accessible.

File: mo_gymnasium/envs/mujoco/hopper_v5.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        EzPickle.__init__(self, cost_objective, **kwargs)
+        self.cost_objetive = cost_objective
+        self.reward_dim = 3 if cost_objective else 2
+        self.reward_space = Box(low=-np.inf, high=np.inf, shape=(self.reward_dim,))
+
+    def step(self, action):
+        observation, reward, terminated, truncated, info = super().step(action)
+        x_velocity = info["x_velocity"]
+        height = 10 * info["z_distance_from_origin"]
+        energy_cost = np.sum(np.square(action))
```
Comment (Kallinteris-Andreas) [inline mo_gymnasium/envs/mujoco/hopper_v5.py:33]: Is there a reason you're computing it here instead of using `info["reward_ctrl"]`

Comment (LucasAlegre) [reply mo_gymnasium/envs/mujoco/hopper_v5.py:33]: I forgot to update this one from v4. Thanks!

File: mo_gymnasium/envs/mujoco/__init__.py [hunk 3]
```diff
@@ -3,56 +3,113 @@
...
+
+
+register(
+    id="mo-ant-v5",
+    entry_point="mo_gymnasium.envs.mujoco.ant_v5:MOAntEnv",
+    max_episode_steps=1000,
+)
+
+register(
+    id="mo-ant-2obj-v5",
```
Comment (ffelten) [inline mo_gymnasium/envs/mujoco/__init__.py:75]: For consistency with v4, I would call it ant-2d-v5

File: mo_gymnasium/envs/mujoco/ant_v5.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    env = mo_gym.make('mo-ant-v4', cost_objective=False)
+    LinearReward(env, weight=np.array([1.0, 0.0]))
+
+    ## Reward Space
+    The reward is 2- or 3-dimensional:
+    - 0: x-velocity
+    - 1: y-velocity
+    - 2: Control cost of the action
+    If the cost_objective flag is set to False, the reward is 2-dimensional, and the cost is added to other objectives.
+    A healthy reward is added to all objectives.
```
Comment (ffelten) [inline mo_gymnasium/envs/mujoco/ant_v5.py:25]: and contact

File: mo_gymnasium/envs/mujoco/half_cheetah_v5.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        EzPickle.__init__(self, **kwargs)
+        self.reward_space = Box(low=-np.inf, high=np.inf, shape=(2,))
+        self.reward_dim = 2
+
+    def step(self, action):
+        observation, reward, terminated, truncated, info = super().step(action)
+        vec_reward = np.array([info["reward_forward"], info["reward_ctrl"]], dtype=np.float32)
```
Comment (ffelten) [inline mo_gymnasium/envs/mujoco/half_cheetah_v5.py:28]: there is no healthy reward here?

File: mo_gymnasium/envs/mujoco/half_cheetah_v5.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+from gymnasium.spaces import Box
+from gymnasium.utils import EzPickle
+
+
+class MOHalfCheehtahEnv(HalfCheetahEnv, EzPickle):
+    """
+    ## Description
+    Multi-objective version of the HalfCheetahEnv environment.
+
+    See [Gymnasium's env](https://gymnasium.farama.org/environments/mujoco/half_cheetah/) for more information.
```
Comment (ffelten) [inline mo_gymnasium/envs/mujoco/half_cheetah_v5.py:12]: I guess this one is Linear([1.0, -0.1]) ?

**Review by ffelten — CHANGES_REQUESTED** [review]
A few comments. Also, I think the visuals of the envs changed in v5, might be worth re-generating the gifs for the documentation.

File: mo_gymnasium/envs/mujoco/half_cheetah_v5.py (hunk 5 above)
Comment (Kallinteris-Andreas) [reply mo_gymnasium/envs/mujoco/half_cheetah_v5.py:28]: No there is not

**Comment by Kallinteris-Andreas** [comment]
The Render behavior was not changed from v4 to v5, only new render arguments were added

**Comment by ffelten** [comment]
> The Render behavior was not changed from v4 to v5, only new render arguments were added

My bad, I thought they became kinda dark mode like this: 
image


Thanks a lot for the support @Kallinteris-Andreas ! Appreciate it :)

File: mo_gymnasium/envs/mujoco/half_cheetah_v5.py (hunk 6 above)
Comment (LucasAlegre) [reply mo_gymnasium/envs/mujoco/half_cheetah_v5.py:12]: Fixed

**Review by ffelten — APPROVED** [review]

**Pull request merged**